src/Containers.jsx

```diff
@@ -314,45 +316,58 @@ class Containers extends React.Component {
 
     render() {
         const columnTitles = [_("Container"), _("CPU"), _("Memory"), _("Owner"), _("State")];
+        const partitionedContainers = { 'no-pod': [] };
+        let filtered = [];
 
         let emptyCaption = _("No containers");
-        if (this.props.containers === null)
-            emptyCaption = _("Loading...");
-        else if (this.props.textFilter.length > 0)
-            emptyCaption = _("No containers that match the current filter");
-        else if (this.props.onlyShowRunning)
+        let emptyCaptionPod = _("No containers in this pod");
+        if (this.props.containers === null || this.props.pods === null) {
+            emptyCaption = emptyCaptionPod = _("Loading...");
+        } else if (this.props.textFilter.length > 0) {
+            emptyCaption = emptyCaptionPod = _("No containers that match the current filter");
+        } else if (this.props.onlyShowRunning)
             emptyCaption = _("No running containers");
-
-        const containersStats = this.props.containersStats;
-        let filtered = [];
-        if (this.props.containers !== null)
-            filtered = Object.keys(this.props.containers).filter(id => !this.props.onlyShowRunning || this.props.containers[id].State == "running");
-
-        if (this.props.ownerFilter !== "all") {
-            filtered = filtered.filter(id => {
-                if (this.props.ownerFilter === "system" && !this.props.containers[id].isSystem)
-                    return false;
-                if (this.props.ownerFilter !== "system" && this.props.containers[id].isSystem)
-                    return false;
-                return true;
-            });
+            emptyCaptionPod = _("No running containers in this pod");
         }
 
-        if (this.props.textFilter.length > 0) {
-            const lcf = this.props.textFilter.toLowerCase();
-            filtered = filtered.filter(id => this.props.containers[id].Names[0].toLowerCase().indexOf(lcf) >= 0 ||
-                    this.props.containers[id].Image.toLowerCase().indexOf(lcf) >= 0
-            );
-        }
+        if (this.props.containers !== null && this.props.pods !== null) {
+            if (this.props.containers !== null && this.props.pods !== null)
+                filtered = Object.keys(this.props.containers).filter(id => !this.props.onlyShowRunning || this.props.containers[id].State == "running");
```

Comment:
I think here is some misunderstanding.
pod is owned by root or system and is running/stopped.

We want to show owner (fine as followup of course, we may also think about not showing owner for separate containers if they are in pod owner by root/user). That means we want to hide the whole user pod if we switch filter from `all` to `system` and the other way around and not just containers inside it. (Because it leaves "empty" pod).

Pod can be running or stopped. We want to show that, fine as followup ofc. That of course means that the filter for running should not filter running containers in pods, but pods as whole. If pod is running but has one died container inside it, we should show also this stopped container (imo).

Reply:
Good point for the owner. I fixed it. I can in a follup remove the owner state from the container columns and move it up to the pod details.

Regarding the state I though it's not so straightforward.
You can have a start/stop a while pod, but you can also stop once container in a pod while all the others are running.

Reply:
> Regarding the state I though it's not so straightforward.
You can have a start/stop a while pod, but you can also stop once container in a pod while all the others are running.

Yeah, but pod being the higher abstraction, I think most filters should apply to it. It might be confusing showing pod with 2 running containers, even though there are 5 containers in the pod. It feels misleading. Maybe if we could indicate somehow that there are additional containers that are hidden for some reason? (Or show number of containers in details).

@garrett WDYT?

Reply:
I would rather have more fine grained filters like demoonstrated here https://www.patternfly.org/2020.06/documentation/react/demos/filtertabledemo and get rid of this logic.

Reply:
That is one approach for sure. It just feels a bit too heavy. I think user just wants to quickly find some container to check if it is running or some image to run it. Going to specific panel, picking column, searching... instead of just typing on the top of the page. But of course, that is just my idea and I want to keep it simple, but maybe it is a bit heavier that I think of it :)

Reply:
@marusak I added the running filters for the pods in the end as you suggested.